src/pages/Account/AccountPayment.tsx

```diff
@@ -1,10 +1,96 @@
+  const [modal, setModal] = React.useState<ModalType | undefined>(undefined);
+  const [alert, setAlert] = React.useState<AlertProperties>({ msg: '', color: '', show: false });
+  const [paymentSource, setPaymentSource] = React.useState<PaymentSource | undefined>(undefined);
+  
+  return (
+    <Query query={GET_PAYMENT_SOURCES}>
+      {({ loading, error, data }) => {
+        if (loading) {
+          return <AudioLoading height={48} width={96} />;
+        }
+        if (error || !data) {
+          return <h4>{error ? error.message : 'Error / No Data'}</h4>;
+        }
+        const paymentSources = data.getPaymentSources;
+        const renderPaymentSources = paymentSources.map((paymentSource: PaymentSource) => (
+          <ListGroupItem
+            key={paymentSource.id}
+            id={paymentSource.id}
+            className="w-100 d-flex justify-content-between align-items-center"
+          >
+            <h6 className="mb-0">
+              {paymentSource.stripeBrand}&nbsp;(...{paymentSource.stripeLast4})
+            </h6>
+            <i onClick={() => handleModal(ModalType.DELETE_CARD, paymentSource)} className="fas fa-trash-alt" />
+          </ListGroupItem>
+        ));
+
+        return (
+          <section>
+            <Alert color={alert.color} isOpen={!!alert.show} toggle={() => setAlert({ ...alert, show: !alert.show })}>{alert.msg}</Alert>
+
+            <Row>
+              <Col xs="12">
+                <p>Credit Balance: {creditBalance.amountUsd}</p>
+              </Col>
+            </Row>
+            <h4>Credit Cards:</h4>
+            <ListGroup className="mb-2 d-flex flex-column">{renderPaymentSources}</ListGroup>
+
+            <Row>
+              <Col xs="12">
+                <div onClick={() => handleModal(ModalType.ADD_NEW_CARD)} className="w-auto d-inline-block align-items-center">
+                  <i className="fas fa-plus-circle" />
+                  <h6 className="ml-2 mb-0 d-inline-block">Add New Card</h6>
+                </div>
+              </Col>
+            </Row>
+
+            <Modal isOpen={modal === ModalType.ADD_NEW_CARD} toggle={handleModal}>
+              <ModalHeader>Add New Card</ModalHeader>
+              <ModalBody>
+                <NewCardForm handleModal={handleModal} setAlert={setAlert} />
+              </ModalBody>
+            </Modal>
+
+            <Modal isOpen={modal === ModalType.DELETE_CARD} toggle={handleModal}>
+              <ModalHeader>Delete Card</ModalHeader>
+              <ModalBody>
+                <DeleteCardForm handleModal={handleModal} paymentSource={paymentSource} setAlert={setAlert} />
+              </ModalBody>
+            </Modal>
+          </section>
+        );
+      }}
+    </Query>
+  );
+
+  function handleModal(modal?: ModalType, newPaymentSource?: PaymentSource) {
+    setModal(modal)
+    newPaymentSource && setPaymentSource(newPaymentSource); // null check modal throws undefined error since fade out takes time
```

Comment:
i think we can remove the fadeOut for now and worry about transitions later

Reply:
just to recap:

When `Delete Card` button has been clicked and processed, the modal initiates its fadeOut. However, the `paymentSource` prop provided to the modal is now `undefined` because that card has been deleted. This crashes the app because the modal tries to display `paymentSource.stripeLast4` while fading out.

There are a couple ways to keep it from crashing. (1) have a mock 'placeholder' paymentSource object to fallback to but the modal will look different while its fading out. (2) keep the deleted paymentSource in state (this may cause a memory leak or some sort because technically the card doesn't exist anymore.

Will conditionally render these modals and compromise fadeOut / fadeIn for simplicity sake